test/new-e2e/tests/ndm/snmp/snmp_test.go

```diff
@@ -155,3 +159,40 @@ func (s *snmpDockerSuite) TestSnmp() {
 		assert.Contains(c, metrics, "snmp.sysUpTimeInstance", "metrics %v doesn't contain snmp.sysUpTimeInstance", metrics)
 	}, 5*time.Minute, 10*time.Second)
 }
+
+func (s *snmpDockerSuite) TestSnmpTagsAreStoredOnRestart() {
+	fakeintake := s.Env().FakeIntake.Client()
+	initialMetrics, err := fakeintake.FilterMetrics("snmp.device.reachable",
```

Comment:
To be safe from potentially flaky test failures I think FIlterMetrics() should be inside an EventuallyWithT().

Reply:
changed https://github.com/DataDog/datadog-agent/pull/26003/commits/d8de7c0722c746a8da7036070e1c7fce2aaa4c8f
also changed in that commit the `require` to `assert` to prevent the test from failing if the fakeintake responds an error